doc/cookbook/multi_node_cluster.md

```diff
@@ -0,0 +1,135 @@
+
+```bash
+export HOST1=<host1-IP>
+export HOST2=<host2-IP>
+export HOST3=<host3-IP>
+```
+
+Add environment variables to each machine:
+
+Start the first instance at the first machine
+```bash
+easegress-server \
+  --cluster-name "multi-node-cluster" \
+  --cluster-role "writer" \
+  --name "machine-1" \
+  --api-addr $HOST1:2381 \ # or 0.0.0.0:2381 if running in docker
+  --cluster-initial-advertise-peer-urls http://$HOST1:2380 \
+  --cluster-listen-peer-urls http://$HOST1:2380 \
+  --cluster-listen-client-urls http://$HOST1:2379 \
+  --cluster-advertise-client-urls http://$HOST1:2379 \
+  --cluster-join-urls http://$HOST1:2380,http://$HOST2:2378,http://$HOST3:2376
+```
+then the second instance at machine 2
+```bash
+easegress-server \
+  --cluster-name "multi-node-cluster" \
+  --cluster-role "writer" \
+  --name "machine-2" \
+  --cluster-initial-advertise-peer-urls http://$HOST2:2378 \
+  --cluster-listen-peer-urls http://$HOST2:2378 \
+  --cluster-listen-client-urls http://$HOST2:2377 \
+  --cluster-advertise-client-urls http://$HOST2:2377 \
+  --cluster-join-urls http://$HOST1:2380,http://$HOST2:2378,http://$HOST3:2376
+```
+and the last machine 3
+```bash
+easegress-server \
+  --cluster-name "multi-node-cluster" \
+  --cluster-role "writer" \
+  --name "machine-3" \
+  --cluster-initial-advertise-peer-urls http://$HOST3:2376 \
+  --cluster-listen-peer-urls http://$HOST3:2376 \
+  --cluster-listen-client-urls http://$HOST3:2375 \
+  --cluster-advertise-client-urls http://$HOST3:2375 \
+  --cluster-join-urls http://$HOST1:2380,http://$HOST2:2378,http://$HOST3:2376
+```
+
+Now list cluster members
+```bash
+egctl --server $HOST1:2381 member list | grep " name"
+# or if running docker
+# egctl --server 0.0.0.0:2381 member list | grep " name"
+```
+should print 
+```bash
+    name: machine-1
+    name: machine-2
+    name: machine-3
+```
+
+###  Add new node
+
+Let's add one more node; this time a reader cluster role.
+
+```bash
+# on a new machine
+export HOST1=<host1-IP>
+export HOST2=<host2-IP>
+export HOST3=<host3-IP>
+export HOST4=<host4-IP>
+
+easegress-server \
+  --cluster-name "multi-node-cluster" \
+  --cluster-role "reader" \
+  --name "machine-4" \
+  --cluster-initial-advertise-peer-urls http://$HOST4:2374 \
+  --cluster-listen-peer-urls http://$HOST4:2374 \
+  --cluster-listen-client-urls http://$HOST4:2373 \
+  --cluster-advertise-client-urls http://$HOST4:2373 \
+  --cluster-join-urls http://$HOST1:2380,http://$HOST2:2378,http://$HOST3:2376,http://$HOST4:2374
+```
+
+
+##  Readers and writers
+
+When running Easegress as a cluster, each instance has either *writer* or *reader* role. There should be odd number off *writers* (1,3,5,7,9) for [failure tolerance](https://etcd.io/docs/v3.5/faq/#what-is-failure-tolerance) and any number of *readers*.
```

Comment:
we need to clarify the relationship between Easegress roles and Etcd roles, or people may get confused why we reference an Etcd document here.

Reply:
And all the examples above have only one `writer`, propose to add an example for multiple writers.